rush/rush/src/actions/InstallAction.ts

```diff
@@ -140,6 +159,70 @@ export default class InstallAction extends CommandLineAction {
     console.log(os.EOL + 'Next you should probably run: "rush link"');
   }
 
+  private _checkThatTempModulesMatch(): void {
+    // read all temp_modules
+    const tempModules: { [packageName: string]: ITempModuleInformation } = {};
+
+    this._tempModulesFiles.forEach((filename) => {
+      const contents: IPackageJson = require(filename);
+      tempModules[contents.name] = {
+        packageJson: contents,
+        existsInProjectConfiguration: false,
+        filename: filename
+      };
+    });
+
+    // 1st ensure that every temp_module exists in the configuration
+    Object.keys(tempModules).forEach((tempModuleName: string) => {
+      const tempModule: ITempModuleInformation = tempModules[tempModuleName];
+
+      let foundMatchingProject: boolean;
+      this._rushConfiguration.projects.forEach((project: RushConfigurationProject) => {
+        if (project.tempProjectName === tempModuleName) {
+          foundMatchingProject = true;
+        }
+      });
+
+      if (!foundMatchingProject) {
+        throw new Error(`The file "${tempModule.filename}" exists in the temp_modules folder ` +
+          `but we could not find a matching project for it. This file may need to be deleted.` +
+          `\n\nDid you forget to run 'rush generate'?`);
+      }
+    });
+
+    // 2nd ensure that each config project has a temp_module which matches the expected value
+    this._rushConfiguration.projects.forEach((project: RushConfigurationProject) => {
+      //   if no temp_module, throw
+
+      const tempModule: ITempModuleInformation
+        = tempModules[project.tempProjectName];
+      if (!tempModule) {
+        throw new Error(`The project ${project.packageName} is missing a corresponding ` +
+          `file in the temp_modules folder.` +
+          `\n\nDid you forget to run 'rush generate'?`);
+      }
+
+      // Generate an expected temp_modules package.json & compare
+      const expectedTempModule: IPackageJson = project.generateTempModule(this._rushConfiguration);
+
+      if (!_.isEqual(expectedTempModule, tempModule.packageJson)) {
+        // this is a workaround for the wordwrap trimming
+        const delimiter: string = String.fromCharCode(129) + String.fromCharCode(129);
+
+        const errorText: string[] = [];
+
+        errorText.push(``);
+        errorText.push(`EXPECTED:\n${JSON.stringify(expectedTempModule, undefined, delimiter)}`);
+        errorText.push(`ACTUAL: \n${JSON.stringify(tempModule.packageJson, undefined, delimiter)}`);
+
+        throw new Error(`The project ${project.packageName}'s ` +
+          `temp_module is outdated:\n` +
+          errorText.join('\n') +
```

Comment:
It will just contain the JSON... The alternate is either not saying anything at all (which is a bit confusing IMO), or to do analysis on the package.json to determine what changed.